JS/Lesson17/sinan/script.js

```diff
@@ -134,38 +134,84 @@ console.log('----------------------------');
 // Input: ({name: "John", age: 30}, "city", "New York")
 // Output: {name: "John", age: 30, city: "New York"}
 
+function getKeyValue (obj,key,value) {
+if(obj[key]=value) return obj;
+} 
+
+const newValue = getKeyValue({name: "John", age: 30}, "city", "New York");
+console.log('Exercise 10:',newValue);
+
 console.log('----------------------------');
 // 11. Write a function that takes an object and a key, and removes the key-value pair from the object.
 // Input: ({name: "John", age: 30, city: "New York"}, "city")
 // Output: {name: "John", age: 30}
 
+function removeKeyValue(testObject, key) {
+  if (testObject.hasOwnProperty(key)) {
+    delete testObject[key];
+    return testObject;
+  } 
+}
+  const ourList = removeKeyValue({name: "John", age: 30, city: "New York"}, "city");
+  console.log('Exercise 11:',ourList);
+
 console.log('----------------------------');
 // 12. Write a function that takes two arrays of equal length and returns an object with the first array as keys and the second array as values.
 // Input: (["name", "age", "city"], ["John", 30, "New York"])
 // Output: {name: "John", age: 30, city: "New York"}
+function concatKeyValue(list1,list2,) {
+  const unifiedList = [];
+  for(let i = 0; i<list1.length; i++){
+    unifiedList[list1[i]] = list2[i];
+  }
+  return unifiedList;
+}
+  const twoList  = concatKeyValue(["name", "age", "city"], ["John", 30, "New York"]);
+  console.log('Exercise 12:',twoList);
 
 // 13. Array Destructuring
 // Given the following array, use array destructuring to assign the values of x, y, and z to their respective variables: 
 const coordinates = [10, 20, 30];
+let chars = ['x','y','z'];
+function concatTwoList(list1,list2) {
```

Comment:
This is not what's asked in the exercise, what you need to do is the following:
`const [x, y, z] = coordinates;`. Also, you don't need `let chars = ['x','y','z'];`